sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/MessagePumpTerminatedException.java

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.servicebus;
+
+
+import com.azure.core.util.logging.ClientLogger;
+
+import static com.azure.core.amqp.implementation.ClientConstants.ENTITY_PATH_KEY;
+import static com.azure.core.amqp.implementation.ClientConstants.FULLY_QUALIFIED_NAMESPACE_KEY;
+import static com.azure.core.amqp.implementation.ClientConstants.PUMP_ID_KEY;
+
+/**
+ * The exception emitted by the mono returned from the {@link SessionsMessagePump#begin()} and
+ * {@link NonSessionMessagePump#begin()} API. If available, the {@link MessagePumpTerminatedException#getCause()}}
+ * indicates the cause for the termination of message pumping.
+ */
+final class MessagePumpTerminatedException extends RuntimeException {
+    private final long pumpId;
+    private final String fqdn;
+    private final String entityPath;
+
+    /**
+     * Instantiate {@link MessagePumpTerminatedException} representing termination of a MessagePump.
+     *
+     * @param pumpId The unique identifier of the MessagePump that terminated.
+     * @param fqdn The FQDN of the host from which the message was pumping.
+     * @param entityPath The path to the entity within the FQDN streaming message.
+     * @param detectedAt debug-string indicating where in the async chain the termination occurred or identified.
+     */
+    MessagePumpTerminatedException(long pumpId, String fqdn, String entityPath, String detectedAt) {
```

Comment:
nit: Fully qualified namespace (fullyQualifedNamespace) rather than fqdn.  It's easy now to remember, but someone new may not get it.

Reply:
Good point, let's use "fullyQualifedNamespace"